contracts/core/Globals.sol

```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+struct Context {
```

Comment:
Also, wouldn't this struct conflict with other structs if users have `Context` defined in their project?

Maybe it's better to have some common interface like `IAssetF`?